web/client/utils/LayersUtils.js

```diff
@@ -33,58 +53,53 @@ var LayersUtils = {
         let mapLayers = configLayers.map((layer) => assign({}, layer, {storeIndex: i++}));
         let groupNames = mapLayers.reduce((groups, layer) => {
             return groups.indexOf(layer.group || 'Default') === -1 ? groups.concat([layer.group || 'Default']) : groups;
-        }, []).filter((group) => group !== 'background');
-        return groupNames.map((group) => {
-            let groupName = group || 'Default';
+        }, []).filter((group) => group !== 'background').reverse();
 
-            return assign({}, {
-                id: groupName,
-                name: groupName,
-                title: groupName,
-                nodes: mapLayers.filter((layer) => (layer.group || 'Default') === group).map((layer) => layer.id).reverse(),
-                expanded: true
-            });
-        }).reverse();
+        return groupNames.reduce((groups, names)=> {
+            let name = names || 'Default';
+            name.split('.').reduce((subGroups, groupName, idx, array)=> {
+                const groupId = name.split(".", idx + 1).join('.');
+                let group = getGroup(groupId, subGroups);
+                const addLayers = (idx === array.length - 1);
+                if (!group) {
+                    group = createGroup(groupId, groupName, mapLayers, addLayers);
+                    subGroups.push(group);
+                }else if (addLayers) {
+                    group.nodes = group.nodes.concat(getLayersId(groupId, mapLayers));
+                }
+                return group.nodes;
+            }, groups);
+            return groups;
+        }, []);
     },
 
     reorder: (groups, allLayers) => {
         return allLayers.filter((layer) => layer.group === 'background')
-            .concat(initialReorderLayers(groups, allLayers));
+            .concat(reorderLayers(groups, allLayers));
     },
     denormalizeGroups: (allLayers, groups) => {
-
         let getGroupVisibility = (nodes) => {
-            let visibility = true;
+            let visibility = false;
```

Comment:
I don't think this refactor is correct. The group visibility is true only if every node is visible